Sources/MpcNET/Commands/Database/FindAndSearchCommands.cs

```diff
@@ -7,10 +7,244 @@
 namespace MpcNET.Commands.Database
 {
     using System.Collections.Generic;
+    using System.ComponentModel;
     using System.Linq;
     using MpcNET.Tags;
     using MpcNET.Types;
 
+    /// <summary>
+    /// Operators available for filters in the protocol
+    /// </summary>
+    public enum FilterOperator
+    {
+        /// <summary>
+        /// Equal (==)
+        /// </summary>
+        [Description("==")]
+        Equal,
+        /// <summary>
+        /// Different (!=)
+        /// </summary>
+        [Description("!=")]
+        Different,
+        /// <summary>
+        /// Contains (contains)
+        /// </summary>
+        [Description("contains")]
+        Contains,
+        /// <summary>
+        /// Mask (=~)
+        /// </summary>
+        [Description("=~")]
+        Mask,
+        /// <summary>
+        /// None ("")
+        /// </summary>
+        [Description("")]
+        None,
+    }
+
+    /// <summary>
+    /// </summary>
+    public static class EnumHelper
+    {
+        /// <summary>
+        /// <see cref="GetDescription"/> of the enum
+        /// </summary>
+        /// <param name="enumValue"></param>
+        public static string GetDescription<T>(this T enumValue)
+            where T : struct, System.IConvertible
+        {
+            if (!typeof(T).IsEnum)
+                return null;
+
+            string description = enumValue.ToString();
+            System.Reflection.FieldInfo fieldInfo = enumValue.GetType().GetField(enumValue.ToString());
+
+            if (fieldInfo != null)
+            {
+                object[] attributes = fieldInfo.GetCustomAttributes(typeof(DescriptionAttribute), true);
+                if (attributes != null && attributes.Length > 0)
+                    description = ((DescriptionAttribute)attributes[0]).Description;
+            }
+
+            return description;
+        }
+    }
+
+    /// <summary>
+    /// Filter interface specialized by filters in the protocol
+    /// </summary>
+    public abstract class IFilter
```

Comment:
This class (and all its related implementations) should be split into separate files in MpcNET/Types:

```
-Types
  |_IFilter.cs
  |_Filters
     |_FilterFile.cs
     |_FilterTag.cs
     |_(...)
  
```